ctru-rs/src/gfx.rs

```diff
@@ -36,6 +54,21 @@ pub trait Screen {
 #[non_exhaustive]
 pub struct TopScreen;
 
+/// A helper container for both sides of the top screen.
+pub struct TopScreen3D<'a> {
+    // morally, this should be &mut or RefMut, but if we do
+    // - &mut: it means gfx can no longer be borrowed immutably while this exists
+    // - RefMut: we don't have an easy way to obtain Ref<dyn Screen> for the left side.
+    //      maybe this one isn't as important since the use case is typically RefMut anyway.
+    //      we could just return &dyn Screen instead of Ref<dyn Screen> ?
+    left: &'a RefCell<TopScreen>,
+    right: RefCell<TopScreenRight>,
+}
+
+// TODO: it feels a little weird to have an asymmetric separate type like this,
+// but maybe if it's not `pub` it's not as weird...
+struct TopScreenRight;
```

Comment:
I think this is OK, since we just hide it behind `dyn Screen`. Alternatively I guess `TopScreen` could hold a `TopScreenLeft` and `TopScreenRight`, and then `TopScreen3D` would have references to each.